tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -129,17 +130,26 @@ public static void main(String[] args) throws Exception {
                 }
             }
 
-            // Make sure all messages are sent before printing out the stats and the metrics
-            producer.flush();
+            if (!shouldPrintMetrics) {
+                producer.close();
 
-            /* print final results */
-            stats.printTotal();
+                /* print final results */
+                stats.printTotal();
+            } else {
+                // Make sure all messages are sent before printing out the stats and the metrics
+                // We need to use reflection for now since tests/kafkatest/sanity_checks/test_performance_services.py
+                // expects the class to bind with old version of client jars.
+                Class c= Class.forName("org.apache.kafka.clients.producer.KafkaProducer");
+                Method flush = c.getDeclaredMethod("flush");
+                flush.invoke(producer);
```

Comment:
I don't think we need this because we always compile against the current version. The runtime error won't happen unless `shouldPrintMetrics` is true.